src/com/serial/qserial_state.py

```diff
@@ -0,0 +1,174 @@
+        """This method checks if the serial is disconnected in the missing state
+
+        Returns:
+            bool: True if the serial is disconnected in the missing state, False otherwise
+        """
+        return self._check_disconnect_condition() and self._reconnecting_attempts == 0
+
+    def _check_missing_condition(self) -> bool:
+        """This method checks if the serial is missing
+
+        Returns:
+            bool: True if the serial is missing, False otherwise
+        """
+        ports = [port.name for port in self._serial.get_available_ports()]
+        return self._serial.port not in ports
+
+    def _connected_state_routine(self):
+        """This method is called by state_control loop when serial is in the connected state"""
+        if self._check_disconnect_condition():
+            self.change_state(QSerialState.DISCONNECTED)
+        elif self._check_missing_condition():
+            self.change_state(QSerialState.MISSING)
+
+    def _reconnect_to_missing_port(self):
+        """This method tries to reconnect to the missing port,
+        It can raise an exception if the port is not available
+        """
+        missing_port = self._serial.port
+        if self._serial.is_connected():
+            # release the port resources before reconnecting
+            self._serial.disconnect()
+
+        self._serial.connect(missing_port)
+        self.change_state(QSerialState.CONNECTED)
+
+    def _missing_state_routine(self):
+        """This method is called by state_control loop when serial is in the missing state"""
+        if not self._check_missing_condition():
+            try:
+                self._reconnecting_attempts += 1  # TODO: add a limit for the attempts
+                self._reconnect_to_missing_port()
+            except Exception as exc:
+                # workaround: on error try again, sometimes the serial raises
+                # an error even if the port is available, the second try should works
+                logger.warning(f"Can't reconnect to the missing port: {exc}")
+            else:
+                self._reconnecting_attempts = 0
+
+        elif self._check_disconnect_in_missing():
+            self.change_state(QSerialState.DISCONNECTED)
+
+    def _disconnected_state_routine(self):
+        """This method is called by state_control loop when serial is in the disconnected state"""
+        if self._check_connect_condition():
+            self.change_state(QSerialState.CONNECTED)
+
+    def run(self) -> None:
+        """This method runs the thread"""
+        while not self._thread_stop.occurs():
+            # this gives immidiately check for the stop signal, after handling the state
+            QThread.msleep(self.THREAD_SLEEP_MS)
+
+            match self._state:
+                case QSerialState.CONNECTED:
+                    self._connected_state_routine()
+                case QSerialState.MISSING:
+                    self._missing_state_routine()
```

Comment:
you could just add here a check for missing attempts >= some number and some sort of reset function. Noticed that it's in TODO but seems quite important to me especially if we have this untested

Reply:
It is an additional feature, if we see that this does not reconnect automatically, we can do it manually. 
I will leave this as TO DO because of the deadlines.